persistence/src/main/java/edu/unc/lib/dl/persist/services/edit/EditTitleService.java

```diff
@@ -0,0 +1,199 @@
+import io.dropwizard.metrics5.Timer;
+import org.apache.commons.io.IOUtils;
+import org.jdom2.Document;
+import org.jdom2.Element;
+import org.jdom2.input.SAXBuilder;
+import org.jdom2.input.sax.XMLReaderSAX2Factory;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+
+import static edu.unc.lib.dl.xml.JDOMNamespaceUtil.MODS_V3_NS;
+
+/**
+ * Service that manages editing of the mods:title property on an object
+ *
+ * @author smithjp
+ *
+ */
+public class EditTitleService {
+
+    private AccessControlService aclService;
+    private MODSValidator modsValidator;
+    private OperationsMessageSender operationsMessageSender;
+    private RepositoryObjectLoader repoObjLoader;
+
+    private static final Timer timer = TimerFactory.createTimerForClass(EditTitleService.class);
+
+    public EditTitleService() {
+    }
+
+    /**
+     * Changes an object's mods:title
+     *
+     * @param agent security principals of the agent making request
+     * @param pid the pid of the object whose label is to be changed
+     * @param title the new label (dc:title) of the given object
+     */
+    public void editTitle(AgentPrincipals agent, PID pid, String title) throws Exception {
+        try (Timer.Context context = timer.time()) {
+
+            aclService.assertHasAccess(
+                    "User does not have permissions to edit titles",
+                    pid, agent.getPrincipals(), Permission.editDescription);
+
+            ContentObject obj = (ContentObject) repoObjLoader.getRepositoryObject(pid);
+            BinaryObject mods = obj.getDescription();
+
+            String username = agent.getUsername();
+
+            String newMods;
+            InputStream modsStream;
+            try {
+                modsStream = mods.getBinaryStream();
+            } catch (NullPointerException e) {
+                modsStream = null;
+            }
+
+            if (modsStream != null) {
+                String modsString = IOUtils.toString(modsStream, StandardCharsets.UTF_8);
+                ByteArrayInputStream modsByteArray = new ByteArrayInputStream(modsString.getBytes());
+                SAXBuilder sb = new SAXBuilder(new XMLReaderSAX2Factory(false));
+                Document document = sb.build(modsByteArray);
+                Element rootEl = document.getRootElement();
+
+                String oldTitle = getOldTitle(rootEl);
+                if (oldTitle != "no mods:title") {
+                    newMods = updateTitle(document, title);
+                } else {
+                    newMods = addTitleToMODS(document, title);
+                }
+            } else {
+                Document document = new Document();
+                document.addContent(new Element("mods", MODS_V3_NS));
+                newMods = addTitleToMODS(document, title);
+            }
+
+            InputStream newModsStream = new ByteArrayInputStream(newMods.getBytes());
```

Comment:
I'd recommend having `updateTitle` and `addTitleToMODS` return a Document rather than the result of `.toString()` (my understanding is that representation is intended for debugging), and pass the result through `XMLOutputter.outputString` like you are in the test.